sklearn/preprocessing/_target_encoder.py

```diff
@@ -396,42 +417,64 @@ def _fit_encodings_all(self, X, y):
                 y,
                 n_categories,
                 self.target_mean_,
+                sample_weight,
             )
         else:
             encodings = self._fit_encoding_binary_or_continuous(
                 X_ordinal,
                 y,
                 n_categories,
                 self.target_mean_,
+                sample_weight,
             )
         self.encodings_ = encodings
 
         return X_ordinal, X_known_mask, y, n_categories
 
     def _fit_encoding_binary_or_continuous(
-        self, X_ordinal, y, n_categories, target_mean
+        self, X_ordinal, y, n_categories, target_mean, sample_weight=None
     ):
         """Learn target encodings."""
         if self.smooth == "auto":
-            y_variance = np.var(y)
+            if sample_weight is None:
+                y_variance = np.var(y)
+            else:
+                y_variance = np.sum(sample_weight * (y - target_mean) ** 2) / (
+                    np.sum(sample_weight)
+                )
+
+            # Maybe leave `sample_weight` as None so as to make
+            # _target_encoder_fast.pyx more efficient
+            if sample_weight is None:
+                sample_weight = np.ones_like(y)
+
             encodings = _fit_encoding_fast_auto_smooth(
                 X_ordinal,
                 y,
+                sample_weight,
                 n_categories,
                 target_mean,
                 y_variance,
             )
         else:
+            # Maybe leave `sample_weight` as None so as to make
+            # _target_encoder_fast.pyx more efficient
+            if sample_weight is None:
+                sample_weight = np.ones_like(y)
```

Comment:
Here again, `sklearn.utils.validation._check_sample_weight`. But this should probably be done in the public API and then make `sample_weight` a non-optional parameter of `_fit_encoding_binary_or_continuous`.